lib/maths/CCalendarComponentAdaptiveBucketing.cc

```diff
@@ -328,26 +289,25 @@ void CCalendarComponentAdaptiveBucketing::refresh(const TFloatVec& endpoints) {
     // that is equal to the number of points they will receive in one
     // period.
     double count{0.0};
-    for (const auto& value : values) {
+    for (const auto& value : newValues) {
         count += CBasicStatistics::count(value);
     }
-    count /= (endpoints[m] - endpoints[0]);
+    count /= (oldEndpoints[m] - oldEndpoints[0]);
```

Comment:
any possibility this can become 0?

Reply:
In practice no, this is full period being modelled which in this case is always 1 day in seconds. We also rely on the fact that we force the individual endpoints to be at least the data bucketing interval apart.

Arguably, we could make all this code safe, i.e. if things went wrong we'd end up in 0/0 situations which we'd want to treat as one. I think on balance though I'd prefer to keep the code simpler and simply assume this invariant holds.